# HG changeset patch
# Parent b11c73b6a202b9556e4cefdd290b187551619c08

Make EmitBundle(Lock|Unlock|AlignStart|AlignEnd) into small asserting routines.
This is needed because of an upstream change which added new uses of MCStreamer
Keep local changes bundled together.

diff -r b11c73b6a202 include/llvm/MC/MCStreamer.h
--- a/include/llvm/MC/MCStreamer.h	Tue May 31 12:33:49 2011 -0700
+++ b/include/llvm/MC/MCStreamer.h	Mon Jul 11 09:37:43 2011 -0700
@@ -414,18 +414,18 @@
 
     /// EmitBundleLock - Begin a group of instructions which cannot
     /// cross a bundle boundary.
-    virtual void EmitBundleLock() = 0;
+    virtual void EmitBundleLock() { assert(0 && "EmitBundleLock not supported"); }
 
     /// EmitBundleUnlock - End a bundle-locked group of instructions.
-    virtual void EmitBundleUnlock() = 0;
+    virtual void EmitBundleUnlock() { assert(0 && "EmitBundleUnlock not supported"); };
 
     /// EmitBundleAlignStart - Guarantee that the next instruction or
     /// bundle-locked group starts at the beginning of a bundle.
-    virtual void EmitBundleAlignStart() = 0;
+    virtual void EmitBundleAlignStart() { assert(0 && "EmitBundleAlignStart not supported"); };
 
     /// EmitBundleAlignEnd - Guarantee that the next instruction or
     /// bundle-locked group finishes at the end of a bundle.
-    virtual void EmitBundleAlignEnd() = 0;
+    virtual void EmitBundleAlignEnd() { assert(0 && "EmitBundleAlignEnd not supported"); };
     /// @}
     // @LOCALMOD-END
 
